### PR TITLE
Removed `SAMPLED_OP` flag from `torch_geometric/typing.py`

### PR DESCRIPTION
**Pull request by pbielak**
The `SAMPLED_OP` flag was used some time ago in the softmax implementation (`torch_geometrics/utils/softmax.py`) and with commit: https://github.com/pyg-team/pytorch_geometric/commit/d7e03c71cda21d62bb465984fe9bf4488c6c6a5a its usage was removed. However, the flag is still present in the `torch_geometric/typing.py` file.

### CODE REVIEW

**Comment by rusty1s** [comment]
I think we can keep it IMO. No strong pref.

**Comment by pbielak** [comment]
Also IMO if we don't use a particular code fragment anymore, it should be removed ;)
